core/components/analytics/provider.tsx

```diff
@@ -36,8 +50,41 @@ const getAnalytics = (
   return null;
 };
 
-export function AnalyticsProvider({ channelId, settings, children }: PropsWithChildren<Props>) {
-  const analytics = getAnalytics(channelId, settings);
+export function AnalyticsProvider({
+  channelId,
+  isCookieConsentEnabled,
+  settings,
+  children,
+}: PropsWithChildren<Props>) {
+  const { consents } = useConsentManager();
+  const prevConsentsRef = useRef<Record<string, boolean> | null>(null);
+
+  const analytics = getAnalytics({
+    channelId,
+    isCookieConsentEnabled,
+    settings,
+  });
+
+  // Update consent when user changes preferences
+  useEffect(() => {
+    if (!isCookieConsentEnabled || !analytics) {
+      return;
+    }
+
+    const currentConsents = consents;
+    const prevConsents = prevConsentsRef.current;
+
+    // Check if consents have changed
+    if (prevConsents && JSON.stringify(currentConsents) !== JSON.stringify(prevConsents)) {
+      const consentState = getConsent();
+
+      if (consentState) {
+        analytics.consent.consentUpdated(consentState);
+      }
+    }
+
+    prevConsentsRef.current = currentConsents;
+  }, [isCookieConsentEnabled, analytics, consents]);
```

Comment:
I'm not super happy about this code, but if y'all can think of anything better let me know!